### ISSUE TITLE
let `positive_definite` return a PDMat

### ISSUE THREAD

**Issue by st--** [invenia/ParameterHandling.jl#41]
This would allow for avoiding a Cholesky each time e.g. when parametrising the covariance of an `MvNormal`. Would you be willing to accept the PDMats.jl dependency?

**Comment by willtebbutt** [comment]
Good point. I don't generally tend to make a lot of use of `PDMats.jl`, so I'm not overly keen on making `positive_definite` give me a `PDMat`.

One alternative would be to make use of the `deferred` function, so you could write something like
```julia
my_pdmat = deferred(PDMat, positive_definite, S)
```
where `S` is your happens-to-be-positive-definite matrix, and then when you call `value` on `my_pdmat` you'll get a `PDMat`.

It's not quite as clean as returning a `PDMat`, but would it satisfy your use-case?

**Comment by willtebbutt** [comment]
(One thing I would definitely be up for is improving the docstring to suggest a strategy like this, or whatever other strategy seems useful)
